core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java

```diff
@@ -732,9 +732,14 @@ public NativeImageInvokerInfo build() {
                 if (graalVMVersion.compareTo(GraalVM.Version.VERSION_22_3_0) >= 0) {
                     // For build time information
                     nativeImageArgs.add("-H:+CollectImageBuildStatistics");
-                    nativeImageArgs.add("-H:ImageBuildStatisticsFile=" + nativeImageName + "-timing-stats.json");
+                    if (graalVMVersion.compareTo(GraalVM.Version.VERSION_23_0_0) < 0) {
+                        // Starting with GraalVM 23.0 this option is no longer available, the file used to write the
+                        // statistics is always reports/image_build_statistics.json
+                        nativeImageArgs
+                                .add("-H:ImageBuildStatisticsFile=reports/image_build_statistics.json");
+                    }
                     // For getting the build output stats as a JSON file
-                    nativeImageArgs.add("-H:BuildOutputJSONFile=" + nativeImageName + "-build-output-stats.json");
+                    nativeImageArgs.add("-H:BuildOutputJSONFile=reports/build_output_statistics.json");
```

Comment:
The way the collector works, is that it collects those metrics files across all integration tests. Thus, it relies on a unique name of the stats files. In this case the native image name was chosen. So please keep that if at all possible. Since https://github.com/oracle/graal/pull/5336 the build stats json file also includes the total time in seconds, which was the main motivating factor for using `-H:+CollectImageBuildStatistics -H:ImageBuildStatisticsFile=...` so that we have a way of knowing the total build time.

Therefore I'd propose a change like so:

```
if (graalVMVersion.compareTo(GraalVM.Version.VERSION_22_3_0) >= 0) {
                    // For getting the build output stats as a JSON file
                    nativeImageArgs.add("-H:BuildOutputJSONFile=" + nativeImageName + "-build-output-stats.json");
}
if (graalVMVersion.compareTo(GraalVM.Version.VERSION_22_3_0) >= 0 && graalVMVersion.compareTo(GraalVM.Version.VERSION_23_0_0) < 0) {
                    // For build time information as 22.3 didn't have total build time in the build stats json
                    nativeImageArgs.add("-H:+CollectImageBuildStatistics");
                    nativeImageArgs.add("-H:ImageBuildStatisticsFile=" + nativeImageName + "-timing-stats.json");
}
```

Reply:
Done

Reply:
Thanks!